packages/dds/tree/src/feature-libraries/modular-schema/schemaChangeFamily.ts

```diff
@@ -0,0 +1,278 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { assert } from "@fluidframework/common-utils";
+import { ChangeEncoder, ChangeFamily, JsonCompatibleReadOnly, ProgressiveEditBuilder } from "../../change-family";
+import { ChangeRebaser } from "../../rebase";
+import {
+    FieldSchema,
+    GlobalFieldKey,
+    LocalFieldKey,
+    SchemaData,
+    TreeSchema,
+    TreeSchemaIdentifier,
+} from "../../schema-stored";
+import { AnchorSet, Delta, FieldKey } from "../../tree";
+import { brand, Invariant, Multimap } from "../../util";
+import { FieldChangeHandler, FieldChangeMap, FieldChangeset, NodeChangeset } from "./fieldChangeHandler";
+import { FullSchemaPolicy } from "./fieldKind";
+
+export class SchemaChangeFamily implements
+    ChangeFamily<any, NodeChangeset>,
+    ChangeRebaser<NodeChangeset>,
+    ChangeEncoder<NodeChangeset> {
+    _typeCheck?: Invariant<any> | undefined;
+
+    constructor(
+        readonly rootSchema: TreeSchemaIdentifier | undefined,
```

Comment:
In what circumstances would we have an instance that doesn't know its schema?

Reply:
This is different from the mono-vs-polymorphic cases, right?

Reply:
My thinking here was to potentially support a polymorphic root, but I haven't thought about how useful or important that is.

Reply:
And you're thinking that in the "any" case (i.e., the unconstrained polymorphism case) we'd represent that with `undefined` as opposed to a `TreeSchemaIdentifier` that means "any"?
@CraigMacomber, any (ha!) thoughts about that?

Reply:
documents have root field schema (not tree schema). In the case of stored schema, its stored in the document and in the case of view schema, its part of the schema information there as well. Thus I don't think we should need a root schema here. If we need access to schema information in general here, a root schema is not enough, we should need more than that, but I don't think we should be providing schema to the constructor anyway.

One high level thing we need to decide is how ChangeFamilies relate to stored vs view schema.

If we tie ChangeFamilies to stored schema, then the schema they are working with can change over time (as its edited, or different documents are loaded), so passing it into the constructor would not make sense.
Since Change families are supposed to be stateless, even passing a schema repository into the constructor seems like the wrong approach: we should probably pass in whatever schema collection is relevant when performing each edit? Or only consult schema when validating input to the edit builder?

Somehow, we will have to deal with apps wanting to edit in terms of the view schema, but the edits send need to be in terms of the stored schema. Schematize could do this (and this is the best plan I have right now) by wrapping the edit builder, producing an edit builder (or other editing API) for the output that implements itself in terms of the original edit builder. Thus I think the modular change family should focus on stored schema, and leave any difference between that and view schema up to schematize.

This means that schema are not constants, and should probably not be in fields of change families.